source/bao_hyp/config.rst

```diff
@@ -3,7 +3,103 @@ Bao Configuration File
 
 Overview
 --------
-An overview of the configuration file options (structure overview)
+The Bao hypervisor's configuration is managed through a dedicated configuration
+in the form of a C source file. This section provides an in-depth description
+of the various configuration options available.
+
+The configuration file follows a specific structure, as outlined below:
+
+.. code-block:: c
+
+    #include <config.h>
+
+    // Load guests' image
+    VM_IMAGE(img1_name, "/path/to/vm1/binary.bin");
+    VM_IMAGE(img2_name, "/path/to/vm2/binary.bin");
+
+    struct config config = {
+
+        // Shared memory region configuration
+        .shmemlist_size = N,
+        .shmemlist = (struct shmem[]) {
+            [0] = {/*shared memory config*/,},
+            [1] = {/*shared memory config*/,},
+            ...
+            [N] = {/*shared memory config*/,}
+        },
+
+        // Guests Configuration
+        .vmlist_size = NUM_VMs,
+        .vmlist = {
+            { /* VM 0 Config*/},
+            { /* VM 1 Config*/},
+            ...
+            { /* VM N Config*/},
+        }
+    };
+
+Before the configuration itself, it is necessary to declare the VM images using
+the ``VM_IMAGE`` macro. This macro directly embeds the guest binary file into
+the hypervisor image. Here's an example usage of the ``VM_IMAGE``:
+
+.. code-block:: c
+
+  VM_IMAGE(img_name, "/path/to/VM/binary.bin");
+
+The ``VM_IMAGE`` macro has two parameters:
+
+1. The ``img_name``, an unique identifier associated with the image that will
+   later be used to describe the image running on the VM (see `Guest Image`_);
+
+2. A C string with the guest image's binary file path. It can be either an
+   absolute path or a path relative to the config source file.
+
+The configuration takes the form of a C struct (``struct config``) defining
+name ``config``. This struct containing two distinct lists: (i) a list of
+shared memory regions (``shmemlist``) and (ii) a list of VMs (``vmlist``).
+While the list of shared memory regions is optional and may be omitted from the
+configuration, the list of VMs is mandatory and must include at least 1 VM.
+Additionally, for each list, it is necessary to specify the list size using the
+parameters ``shmemlist_size`` and ``vmlist_size``.
+
+Building Bao Hypervisor
```

Comment:
I would move this section more towards the end of the document. Because we start talking about the configuration structure in the previous section, this feels like changing the subject a bit. We should continua with details of the configuration and then, in the end, talk about the location of the configuration file. Also, for this reason, I would change the name of this section to something like: "Configuration File Placement/Location" or something to do with "Configuration repositories"...